### ISSUE TITLE
I can not connect to rally when I'm using VPN

### ISSUE THREAD

**Issue by Levi55555** [RallyTools/RallyRestToolkitForPython#174]
I'm trying to fetch data from rally1.rallydev.com.  When I use my python script without VPN it works fine but after I connect it drops the following error:
"pyral.context.RallyRESTAPIError: Target Rally host: 'rally1.rallydev.com' non-existent or unreachable"

(I have to use SSO and apikey for the connection)

**Comment by klehman-rally** [comment]
for clarification purposes, if you are not on a VPN you script using pyral can successfully connect to rally1.rallydev.com and retrieve Rally information (or create an artifact or update and artifact), correct?
It is only when you are on a VPN attempting to do the same thing that problems occur, correct?

**Comment by Levi55555** [comment]
Yes, the VPN causes the problem. I can successfully fetch data from rally1.rallydev.com when I'm not using VPN.

**Comment by klehman-rally** [comment]
I'd suggest using something like curl or httpie or a browser REST client extension to see if you can connect successfully through the VPN outside of pyral.  Do you know any info on the VPN, like vendor/product/version/etc.?  I wouldn't think that OS issues are suspect here, but what platform are you running on?
The following is a basic url you can try:
https://rally1.rallydev.com/slm/webservice/v2.0/user?fetch=true&query=(User = "you")
Observe any diagnostic output looking for abnormalities.

**Comment by geekymeek** [comment]
I ran into this and there being a proxy when on the vpn. I used the below to deal with it, set a cmd line arg to use the prox, then set the OS env vars to use it with the correct values...

    parser.add_argument('-p', '--proxy', help='use inet proxy', action='store_true')
....
    # inet proxy
    if (args.proxy):
        user = getpass.getuser()  # from OS env USERNAME var "should" be set
        pwd = getpass.getpass(prompt='Proxy Password:')
        os.environ['HTTP_PROXY'] = 'http://{}:{}@yourproxyserver:8080'.format(user, pwd)
        os.environ['HTTPS_PROXY'] = 'https://{}:{}@yourproxyserver:8080'.format(user, pwd)
        print('using inet proxy')

...
then rally calls should work
